### ISSUE TITLE
Initial conditions in shock decomposition

### ISSUE THREAD

**Issue by AndreyDO** [IRIS-Solutions-Team/IRIS-Toolbox#220]
Hi, guys!

I'm doing a shock decomposition exercise via the command 'simulate'.
Why might the contribution of 'initial conditions & constant' term increase over time while theoretical autocorrelation for the variable that is being decomposed is below unity?

Best,
Andrey Orlov

**Comment by jaromir-benes** [comment]
Hi Andrey

It's increasing because it's a combined contribution of both the
initial condition AND the constant term (which is added every period). The
contribution of the initial condition alone would be decreasing over time
but the constant term is added every period.

Let's say you have an autoregressive equation
x = 0.9*x{-1} + 1 + shock;

If you start your simulation from x=0, then the contribution of the initial
cond+constant term is

1, 1.9, 2.71, 3.439, ..., 10

Note that the contributions will depend on the value of the initial
condition itself - if you instead start at x=10, then the contributions
will remain constant

10, 10, 10, ....

while starting above 10 will produce a declining contribution.

Best
Jaromir

**Comment by AndreyDO** [comment]
Thanks for the clarification, Jaromir!

**Comment by AndreyDO** [comment]
Thanks for the clarification, Jaromir! 
Just one question more :). 
This constant term, is it the same as the steady-state value of the variable?

**Comment by jaromir-benes** [comment]
Not exactly. Using the previous example again, the relationship between the
constant term and the steady state of a variable depends also on the
autoregression

x = rho * x{-1} + const + shock

x_ss = const / (1 - rho)

(assuming rho is stable)

Best
Jaromir

**Comment by AndreyDO** [comment]
Got it! Thanks!

**Comment by ripatti** [comment]
Strangely, I get complex numbers as the contribution of 'initial conditions & constant' term (imaginary part is "large"). Is this related to some trends (there should not be one) or non stationary (there should not be one)?

**Comment by jaromir-benes** [comment]
Strange, indeed. Can you please share the codes (or a simplified variant of
those)?

Thanks
Mirek

**Comment by ripatti** [comment]
Thanks for the prompt reply Mirek!

I forgot to add important info: this concerns Iris version iris20180629
Since this is part of a large codebase (BOF model code environment), I have
not yet been able to test it with newer version. I will try this first and
then try to isolate the problem to a smaller set. I figured out the problem
after changing two data series.

I will come back later.

Cheers,

Antti



su 12. tammik. 2020 klo 14.58 Jaromír Beneš (notifications@github.com)
kirjoitti:

**Comment by jaromir-benes** [comment]
Ok, Antti - I'll wait for you message...

**Comment by ripatti** [comment]
This was my mistake. I manage to recycle my data such that - in the end -
I took log of negative numbers. Embarrassing. Sorry for bothering you.

Best,

Antti

su 12. tammik. 2020 klo 22.13 Jaromír Beneš (notifications@github.com)
kirjoitti:

**Comment by jaromir-benes** [comment]
Happens all the time... :)

**Comment by nfaqs** [comment]
Hi,

Is it possible within IRIS to separate the contribution of initial conditions from the contribution of constants? I checked the tutorials 

https://github.com/IRIS-Solutions-Team/Tutorial-Simple-SPBC-Model/blob/master/run15_filterHistData.m 

and 

https://github.com/IRIS-Solutions-Team/Tutorial-Simple-SPBC-Model/blob/master/run16_moreKalmanFilter.m 

and don't see any documentation of that, but perhaps I missed something.

Thanks!

**Comment by jaromir-benes** [comment]
No, unfortunately not. The main reason is that it would make sense only in
stationary models. In models with one or multiple unit roots or log unit
roots, the initial conditions would have to be expresses in terms of some
underlying deviations of cointegrating combinations of variables, plus
arbitrary terms related to the underlying unit root processes (there are,
in general, infinitely many representations/rotations separating unit-root
processes, corresponding to the infinitely many Schur decompositions).

If you have a stationary model, you can run this manually, by setting up a
number of simulations with replacing all initials with their steady state
values except one at a time.

Best,
Jaromir
